### PR TITLE
Add support for Comet Logging 

### PR DESCRIPTION
**Pull request by sherpan**
# Related Issue or bug

None
Closes #[issue number that will be closed through this PR]

# Describe the changes you've made

Added support for logging to Comet 

# Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

 `# load dataset
from pycaret.datasets import get_data
data = get_data('diabetes')

# init setup
from pycaret.classification import *
clf1 = setup(data, target = 'Class variable', log_experiment = 'comet_ml', experiment_name = 'asset_pycaret', log_data=True),

# model training
best_model = compare_models() `

# Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)

N/A

# Checklist:

- [x ] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

# Screenshots

 Original           | Updated
 :--------------------: |:--------------------:
 **original screenshot**  | <b>updated screenshot </b> |

### CODE REVIEW

File: pycaret/nlp/__init__.py [hunk 1]
```diff
@@ -14,7 +14,12 @@
 from pycaret.loggers.base_logger import BaseLogger
 from pycaret.loggers.dagshub_logger import DagshubLogger
 from pycaret.loggers.mlflow_logger import MlflowLogger
+<<<<<<< HEAD
```
Comment (ngupta23) [inline pycaret/nlp/__init__.py:17]: Looks like there is a conflict. Can you fix this?

Comment (sherpan) [reply pycaret/nlp/__init__.py]: should be fixed now!

File: requirements-optional.txt [hunk 2]
```diff
@@ -31,6 +31,8 @@ fastapi>=0.75.0  # For web api
 uvicorn>=0.17.6  # For web api
 m2cgen>=0.9.0  # For model conversion
 evidently>=0.1.45.dev0  # for drift reporting
+comet_ml>=3.31.22 #comet ml tracking
+joblib >=1.2.0 #file pipeliing
```
Comment (celestinoxp) [inline requirements-optional.txt:35]: joblib exists in requirements.txt, i think you don't need insert insert in requirements-optional.txt

Comment (sherpan) [reply requirements-optional.txt]: deleted!

**Comment by moezali1** [comment]
@sherpan Can you please also provide a colab notebook and demonstrate how the functionality is working?

Thanks.

**Comment by sherpan** [comment]
@moezali1 

I put some example code in the PR but realized the formatting got messed up. Here it is. I can turn this into a colab when stuff is merged! 

```
from pycaret.datasets import get_data
data = get_data('diabetes')

# init setup
from pycaret.classification import *
clf1 = setup(data, target = 'Class variable', log_experiment = 'comet_ml', experiment_name = 'py_caret_diabetes', log_data=True),

# model training
best_model = compare_models() 
```

**Comment by sherpan** [comment]
@celestinoxp @ngupta23  not sure why the code quality is failing. Im pretty sure I have installed the pre-commit hooks and didnt get anything. Any advice on how to re-produce the codestyle errors on my end so i can make sure its passing?

**Comment by Yard1** [comment]
@sherpan Can you run `pip install black==22.12.0 isort==5.11.4` and then run `black pycaret; isort --profile=black pycaret` manually?

**Comment by celestinoxp** [comment]
> @sherpan Can you run `pip install black==22.12.0 isort==5.11.4` and then run `black pycaret; isort --profile=black pycaret` manually?

uma dúvida, porquê essas versões fixas? Não deveria instalar antes os requerimentos de requirements-dev.txt?

**Comment by celestinoxp** [comment]
> @celestinoxp @ngupta23 not sure why the code quality is failing. Im pretty sure I have installed the pre-commit hooks and didnt get anything. Any advice on how to re-produce the codestyle errors on my end so i can make sure its passing?

If you use vscode, you might have to set black to format the code.
Just to confirm, make sure you have the https://github.com/pycaret/pycaret/blob/master/requirements-dev.txt dependencies installed and vscode (or other) configured to black format the code.
If this is the case with VSCode, I solved it with lines in settings.json:   
"editor.defaultFormatter": "ms-python.black-formatter",
"editor.formatOnType": true,
"editor.formatOnSave": true,

**Comment by Yard1** [comment]
@celestinoxp those are the versions we use in CI. I'll pin them in requirements as well.

**Comment by sherpan** [comment]
thanks @celestinoxp @Yard1 for your instructions. I think i fixed it. here is what i got when i added my latest comet 
Screen Shot 2023-02-02 at 4 12 29 PM

**Review by Yard1 — APPROVED** [review]

**Pull request merged**